stdlib/public/Glibc/Glibc.swift

```diff
@@ -29,6 +29,82 @@ public var errno: Int32 {
 // fcntl.h
 //===----------------------------------------------------------------------===//
 
+@warn_unused_result
+@_silgen_name("_swift_Glibc_open")
+func _swift_Glibc_open(path: UnsafePointer<CChar>,
+  _ oflag: CInt, _ mode: mode_t) -> CInt
+
+@warn_unused_result
+@_silgen_name("_swift_Glibc_openat")
+func _swift_Glibc_openat(fd: CInt,
+  _ path: UnsafePointer<CChar>,
+  _ oflag: CInt, _ mode: mode_t) -> CInt
+
+@warn_unused_result
+public func open(path: UnsafePointer<CChar>, _ oflag: CInt) -> CInt {
+  return _swift_Glibc_open(path, oflag, 0)
+}
+
+@warn_unused_result
+public func open(path: UnsafePointer<CChar>, _ oflag: CInt,
+  _ mode: mode_t) -> CInt {
+  return _swift_Glibc_open(path, oflag, mode)
+}
+
+@warn_unused_result
+public func openat(fd: CInt, _ path: UnsafePointer<CChar>,
+  _ oflag: CInt) -> CInt {
+  return _swift_Glibc_openat(fd, path, oflag, 0)
+}
+
+@warn_unused_result
+public func openat(fd: CInt, _ path: UnsafePointer<CChar>,
+  _ oflag: CInt, _ mode: mode_t) -> CInt {
+  return _swift_Glibc_openat(fd, path, oflag, mode)
+}
+
+@warn_unused_result
+@_silgen_name("_swift_Glibc_fcntl")
+internal func _swift_Glibc_fcntl(
+  fd: CInt,
+  _ cmd: CInt,
+  _ value: CInt
+  ) -> CInt
+
+@warn_unused_result
+@_silgen_name("_swift_Glibc_fcntlPtr")
+internal func _swift_Glibc_fcntlPtr(
+  fd: CInt,
+  _ cmd: CInt,
+  _ ptr: UnsafeMutablePointer<Void>
+  ) -> CInt
```

Comment:
No problem, I’m just about done with the changes.  Once done, I’ll do a build to confirm and run my external test program compiled with the new compiler.  When that’s done, I’ll push my changes.


> On Dec 10, 2015, at 9:56 PM, Dmitri Gribenko <notifications@github.com> wrote:
> 
> In stdlib/public/Glibc/Glibc.swift <https://github.com/apple/swift/pull/413#discussion_r47317229>:
> 
> > +
> > +@warn_unused_result
> > +@_silgen_name("_swift_Glibc_fcntl")
> > +internal func _swift_Glibc_fcntl(
> > +  fd: CInt,
> > +  _ cmd: CInt,
> > +  _ value: CInt
> > +  ) -> CInt
> > +
> > +@warn_unused_result
> > +@_silgen_name("_swift_Glibc_fcntlPtr")
> > +internal func _swift_Glibc_fcntlPtr(
> > +  fd: CInt,
> > +  _ cmd: CInt,
> > +  _ ptr: UnsafeMutablePointer<Void>
> > +  ) -> CInt
> Thank you! Patches that fix existing code to conform are also welcome!
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift/pull/413/files#r47317229>.
>